src/hotkey/hotkey_command.cpp

```diff
@@ -36,239 +36,239 @@ namespace {
-	{ HOTKEY_LEADER, "leader", N_("Leader"), false, scope_game, HKCAT_UNITS, "" },
-	{ HOTKEY_UNDO, "undo", N_("Undo"), false, scope_game | scope_editor, HKCAT_GENERAL, "" },
-	{ HOTKEY_REDO, "redo", N_("Redo"), false, scope_game | scope_editor, HKCAT_GENERAL, "" },
-	{ HOTKEY_ZOOM_IN, "zoomin", N_("Zoom In"), false, scope_game | scope_editor, HKCAT_GENERAL, "" },
-	{ HOTKEY_ZOOM_OUT, "zoomout", N_("Zoom Out"), false, scope_game | scope_editor, HKCAT_GENERAL, "" },
-	{ HOTKEY_ZOOM_DEFAULT, "zoomdefault", N_("Default Zoom"), false, scope_game | scope_editor, HKCAT_GENERAL, "" },
-	{ HOTKEY_FULLSCREEN, "fullscreen", N_("Toggle Full Screen"), false, scope_game | scope_editor | scope_main, HKCAT_GENERAL, "" },
-	{ HOTKEY_SCREENSHOT, "screenshot", N_("Screenshot"), false, scope_game | scope_editor | scope_main, HKCAT_GENERAL, "" },
-	{ HOTKEY_MAP_SCREENSHOT, "mapscreenshot", N_("Map Screenshot"), false, scope_game | scope_editor, HKCAT_GENERAL, "" },
-	{ HOTKEY_ACCELERATED, "accelerated", N_("Accelerated"), false, scope_game, HKCAT_GENERAL, "" },
-	{ HOTKEY_TERRAIN_DESCRIPTION, "describeterrain", N_("Terrain Description"), false, scope_game | scope_editor, HKCAT_MAP, "" },
-	{ HOTKEY_UNIT_DESCRIPTION, "describeunit", N_("Unit Description"), false, scope_game | scope_editor, HKCAT_UNITS, "" },
-	{ HOTKEY_RENAME_UNIT, "renameunit", N_("Rename Unit"), false, scope_game | scope_editor, HKCAT_UNITS, "" },
-	{ HOTKEY_DELETE_UNIT, "editor-deleteunit", N_("Delete Unit"), false, scope_game | scope_editor, HKCAT_TOOLS, "" },
-
-	{ HOTKEY_SAVE_GAME, "save", N_("Save Game"), false, scope_game, HKCAT_SAVING, "" },
-	{ HOTKEY_SAVE_REPLAY, "savereplay", N_("Save Replay"), false, scope_game, HKCAT_SAVING, "" },
-	{ HOTKEY_SAVE_MAP, "savemap", N_("Save Map"), false, scope_game, HKCAT_SAVING, "" },
-	{ HOTKEY_LOAD_GAME, "load", N_("Load Game"), false, scope_game | scope_main, HKCAT_SAVING, "" },
-	{ HOTKEY_RECRUIT, "recruit", N_("Recruit"), false, scope_game, HKCAT_UNITS, "" },
-	{ HOTKEY_REPEAT_RECRUIT, "repeatrecruit", N_("Repeat Recruit"), false, scope_game, HKCAT_UNITS, "" },
-	{ HOTKEY_RECALL, "recall", N_("Recall"), false, scope_game, HKCAT_UNITS, "" },
-	{ HOTKEY_LABEL_SETTINGS, "label_settings", N_("Show/Hide Labels"), false, scope_game, HKCAT_MAP, "" },
-	{ HOTKEY_ENDTURN, "endturn", N_("End Turn"), false, scope_game, HKCAT_GENERAL, "" },
+	{ HOTKEY_SCROLL_UP, "scroll-up", N_("Scroll Up"), false, true, scope_game | scope_editor, HKCAT_GENERAL, "" },
+	{ HOTKEY_SCROLL_DOWN, "scroll-down", N_("Scroll Down"), false, true, scope_game | scope_editor, HKCAT_GENERAL, "" },
+	{ HOTKEY_SCROLL_LEFT, "scroll-left", N_("Scroll Left"), false, true, scope_game | scope_editor, HKCAT_GENERAL, "" },
+	{ HOTKEY_SCROLL_RIGHT, "scroll-right", N_("Scroll Right"), false, true, scope_game | scope_editor, HKCAT_GENERAL, "" },
+
+	{ HOTKEY_CANCEL, N_("cancel"), N_("Cancel"), false, false, scope_game | scope_editor | scope_main, HKCAT_GENERAL, "" },
```

Comment:
You're adding a new parameter to so many hotkeys and setting it to a non-default value only for four of them...

Have you considered specifying toggle hotkeys in a different way, e.g. by having a separate array for them?

(That is, one array `hotkey_list_` that contains all hotkeys including toggle hotkeys, just like now, and another array, that contains only the HOTKEY_COMMAND values of toggle hotkeys. On startup, loop over the second array, and for every command there, find the corresponding hotkey in the first array and mark it as a toggle hotkey.)

Reply:
Hmm... let's say that I'd weakly prefer it.